### PR TITLE
Fix remove button to remove selected tags

### PR DESCRIPTION
**Pull request by kurtw**
Creates a hidden field to post the tag data to delete from the post. Clicking on the "X" button next to a tag will remove it from the HTML and then after saving the post will delete the tag data.

### CODE REVIEW

File: inc/meta-box-html.php [hunk 1]
```diff
@@ -459,8 +459,9 @@ protected function date( $taxonomy, $tax_nice_name, $multiples = false, $require
 					if ( is_numeric( $term->name ) ) {
 						$natdate = date( 'j F, Y', intval( $term->name ) );
 	?>
-			  <span><a id='<?php echo sanitize_text_field( $taxonomy ) ?>-check-num-<?php echo sanitize_text_field( $i ) ?>' class='ntdelbutton <?php echo sanitize_text_field( $term->name ) ?>'><?php echo sanitize_text_field( $term->name ) ?></a>&nbsp;<?php echo $natdate ?></span>
+			  <span><a id='<?php echo sanitize_text_field( $taxonomy ) ?>-check-num-<?php echo sanitize_text_field( $i ) ?>' class='datedelbutton <?php echo esc_attr( $term->name ) ?>'><?php echo sanitize_text_field( $term->name ) ?></a><?php echo $natdate ?></span>
```
Comment (Scotchester) [inline inc/meta-box-html.php]: Please re-add the `&nbsp;` lost here when you fixed the merge.

File: inc/meta-box-models.php [hunk 2]
```diff
@@ -303,6 +303,11 @@ public function validate_taxonomyselect($field, $post_ID) {
  */
 
 public function validate_date($field, $post_ID) {
+    $i = 0;
+    while ( isset( $_POST['rm_' . $field['taxonomy'] . '_' . $i ]) ) {
```
Comment (Scotchester) [inline inc/meta-box-models.php]: Wouldn't this loop logic fail if someone tries to remove the second or later term in the list, but not the first?

File: inc/meta-box-html.php [hunk 3]
```diff
@@ -459,8 +459,9 @@ protected function date( $taxonomy, $tax_nice_name, $multiples = false, $require
...
 			<?php
+						HTML::hidden( 'rm_' . $tax_name . '_' . $i, null, null );
 					} else {
 						$date = strtotime( $term->name ); // If it isn't, convert it to a timestamp -- why? ?>
 			<span><a id='<?php echo sanitize_text_field( $taxonomy ) ?>-check-num-<?php echo sanitize_text_field( $i ) ?>' class='ntdelbutton <?php echo sanitize_text_field( $term->name ) ?>'><?php echo sanitize_text_field( $term->name ) ?></a>&nbsp;<?php echo sanitize_text_field( $term->name ) ?></span>
```
Comment (Scotchester) [inline inc/meta-box-html.php]: It doesn't appear to matter for now, but we should probably update this HTML in the same way as line 462, just in case.

Comment (Scotchester) [inline inc/meta-box-html.php]: And while you're in there, in the `class` attribute, change what it's echoing from

```php
<?php echo sanitize_text_field( $term->name ) ?>
```

to

```php
<?php echo sanitize_text_field( $date ) ?>
```

so we get the newly created timestamp, not a date string that includes spaces (which would mess with your `functions.js` line 80).

File: inc/meta-box-html.php (hunk 1 above)
Comment (Scotchester) [inline inc/meta-box-html.php]: As a side note: Why the change to `esc_attr`? It seems like `esc_attr` is used more prevalently in other places in this file. @gboone Any insight into why you chose `sanitize_text_field` here?

Comment (Scotchester) [inline inc/meta-box-html.php]: @kurtw Please address the comments on this line.

File: inc/meta-box-html.php (hunk 3 above)
Comment (Scotchester) [inline inc/meta-box-html.php]: Please also address the second comment on this line.

File: inc/meta-box-html.php (hunk 1 above)
Comment (gboone) [inline inc/meta-box-html.php]: `esc_attr` is [better suited for html attributes](http://codex.wordpress.org/Function_Reference/esc_attr). `sanitize_text_field` is probably a bit overkill for html attributes, I think I learned that while writing the form and changed some but not all of them.

Comment (Scotchester) [inline inc/meta-box-html.php]: Thanks, Greg! Kurt, let's go ahead and change these all to `esc_attr` before merging.

**Comment by Scotchester** [comment]
All good! :+1:
